### PR TITLE
snowmachine: init at 1.0.1

### PR DESCRIPTION
**Pull request by djanatyn**
###### Motivation for this change
Getting into the winter spirit with some snowflakes!

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

When looking for the license, I saw that it was created as BSD, but it didn't mention which BSD license:
```
$ rg 'BSD' -C 2 setup.py
21-    description=("A python script that will make your terminal "
22-                 "snow"),
23:    license = "BSD",
24-    setup_requires=['setuptools-git'],
25-    install_requires=requires,
--
34-        "Environment :: Console",
35-        "Environment :: Console :: Curses",
36:        "License :: OSI Approved :: BSD License",
37-    ],
38-    entry_points = {

```

I marked this as `unfree`. I'm opening an issue to ask the author to describe which BSD license to use.

### CODE REVIEW

**Comment by djanatyn** [comment]
I opened an issue for the licensing upstream: https://github.com/sontek/snowmachine/issues/11

**Comment by sontek** [comment]
👍  I added the LICENSE and the project is tested / works on MacOS as well if that matters.  (also works on windows)

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 105895` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>snowmachine</li>
  </ul>
</details>

**Comment by djanatyn** [comment]
> +1 I added the LICENSE and the project is tested / works on MacOS as well if that matters (also works on windows). I'm not a license expert by any means, if there is a LICENSE you prefer thats better let me know. It was my understanding that BSD was the nicest/easiest to work with so I marked that as the one

That's great! License is definitely your choice, and BSD-3 works great.

The only issue I had was understanding which version of the BSD license. Thank you for clarifying and closing https://github.com/sontek/snowmachine/issues/11 😄

I just pushed `ac13939af3e159a2b5836fd33dd7231d4572602f2` to update the license to `bsd3`:
```diff
diff --git a/pkgs/applications/snowmachine/default.nix b/pkgs/applications/snowmachine/default.nix
index 878d1f494f3..d34f46d7d0d 100644
--- a/pkgs/applications/snowmachine/default.nix
+++ b/pkgs/applications/snowmachine/default.nix
@@ -15,7 +15,7 @@ buildPythonPackage rec {
   meta = with lib; {
     description = "A python script that will make your terminal snow";
     homepage = "http://github.com/sontek/snowmachine";
-    license = with licenses; [ unfree ];
+    license = with licenses; [ bsd3 ];
     maintainers = with maintainers; [ djanatyn ];
   };
 }
```

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 105895` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>snowmachine</li>
  </ul>
</details>

File: pkgs/applications/snowmachine/default.nix [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  version = "1.0.1";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1v385hhxy2a8vx5p0fhn0di8l4qfpb0a86j6nwsg0aw6ngb09qf1";
+  };
+
+  buildInputs = [ setuptools-git ];
+  propagatedBuildInputs = [ click colorama ];
+
```
Comment (SuperSandro2000) [inline pkgs/applications/snowmachine/default.nix:14]:
```suggestion
  doCheck = false;
```

Does the package contain tests? Can we do an pythonImportsCheck?

Comment (djanatyn) [reply pkgs/applications/snowmachine/default.nix:14]: The package does not contain any tests. I added `pythonImportsCheck`:
```diff
commit ae0e0d21cfeaf977b38a316988f23330c948d882
Author: Jonathan Strickland <djanatyn@gmail.com>
Date:   Sun Jan 17 16:06:20 2021 -0500

    snowmachine: add pythonImportsCheck
    
    Co-authored-by: Sandro <sandro.jaeckel@gmail.com>

diff --git a/pkgs/applications/snowmachine/default.nix b/pkgs/applications/snowmachine/default.nix
index d34f46d7d0d..01c8d050fdc 100644
--- a/pkgs/applications/snowmachine/default.nix
+++ b/pkgs/applications/snowmachine/default.nix
@@ -12,6 +12,9 @@ buildPythonPackage rec {
   buildInputs = [ setuptools-git ];
   propagatedBuildInputs = [ click colorama ];
 
+  doCheck = false;
+  pythonImportsCheck = [ "click" "colorama" ];
+
   meta = with lib; {
     description = "A python script that will make your terminal snow";
     homepage = "http://github.com/sontek/snowmachine";

```

File: pkgs/applications/snowmachine/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  doCheck = false;
+  pythonImportsCheck = [ "click" "colorama" ];
```
Comment (SuperSandro2000) [inline pkgs/applications/snowmachine/default.nix:16]:
```suggestion
  pythonImportsCheck = [ "snowmachine" ];
```

It should test the package we are building not a dependency.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 105895` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>snowmachine</li>
</ul>
</details>

**Pull request merged**